stdlib/public/core/Slice.swift.gyb

```diff
@@ -96,13 +97,9 @@ def get_slice_doc_comment(Self):
 %   for Mutable in [ False, True ]:
 %     for RangeReplaceable in [ False, True ]:
 %       Self = sliceTypeName(traversal=Traversal, mutable=Mutable, rangeReplaceable=RangeReplaceable)
-%       BaseRequirements = ['_' + collectionForTraversal(Traversal).replace('Collection', 'Indexable')]
-%       if Mutable:
-%         BaseRequirements.append('_MutableIndexable')
-%       if RangeReplaceable:
-%         BaseRequirements.append('_RangeReplaceableIndexable')
-%       BaseRequirements = ' & '.join(BaseRequirements)
+%       BaseRequirements = ' & '.join(protocolsForCollectionFeatures(traversal=Traversal, mutable=Mutable, rangeReplaceable=RangeReplaceable))
 %       SelfProtocols = ', '.join(protocolsForCollectionFeatures(traversal=Traversal, mutable=Mutable, rangeReplaceable=RangeReplaceable))
+%       Indices = defaultIndicesForTraversal(Traversal)
```

Comment:
I think this resolves `FIXME(ABI)#67` that appears slightly below:

>    // FIXME(ABI)#67 (Associated Types with where clauses):
   //
   //   public typealias Indices = Base.Indices
   //   public var indices: Indices { ... }
   //
   // We can't do it right now because we don't have enough
   // constraints on the Base.Indices type in this context.
 
I don't think that's right: `Slice` can't use `Base.Indices` as its `Indices` because it would have no way of constructing it correctly with the slice's start and end not the base's. At best, it could maybe use `Base.Indices.SubSequence`, and then still implement `var indices` to return `_base.indices[_startIndex..<_endIndex]` but there's not much benefit of this over just using the default `Indices` implementation on `Self`.

Reply:
Oh wait, we constrain `Indices.SubSequence: Indices` so we could do it. I'll try that as a separate PR.